Eventer panics

```
E0205 10:35:30.002817       1 influxdb.go:117] Failed to create infuxdb: failed to ping InfluxDB server at "monitoring-influxdb:8086" - Get http://monitoring-influxdb:8086/ping: EOF
panic: runtime error: invalid memory address or nil pointer dereference
[signal 0xb code=0x1 addr=0x30 pc=0x50d441]

goroutine 26 [running]:
k8s.io/heapster/events/sinks/influxdb.(*influxdbSink).sendData(0xc8201aae80, 0xc82043c000, 0x2, 0xa)
        /usr/local/google/home/pszczesniak/go/src/k8s.io/heapster/events/sinks/influxdb/influxdb.go:126 +0x221
k8s.io/heapster/events/sinks/influxdb.(*influxdbSink).ExportEvents(0xc8201aae80, 0xc82042eea0)
        /usr/local/google/home/pszczesniak/go/src/k8s.io/heapster/events/sinks/influxdb/influxdb.go:111 +0x379
k8s.io/heapster/events/sinks.NewEventSinkManager.func1(0x7fd28255e2b8, 0xc8201aae80, 0xc820328300, 0xc820328360)
        /usr/local/google/home/pszczesniak/go/src/k8s.io/heapster/events/sinks/manager.go:59 +0xe3
created by k8s.io/heapster/events/sinks.NewEventSinkManager
        /usr/local/google/home/pszczesniak/go/src/k8s.io/heapster/events/sinks/manager.go:68 +0x238

goroutine 1 [chan receive]:
main.main()
        /usr/local/google/home/pszczesniak/go/src/k8s.io/heapster/events/eventer.go:88 +0xc44

goroutine 5 [chan receive]:
github.com/golang/glog.(*loggingT).flushDaemon(0x1843c80)
        /usr/local/google/home/pszczesniak/go/src/k8s.io/heapster/Godeps/_workspace/src/github.com/golang/glog/glog.go:879 +0x67
created by github.com/golang/glog.init.1
        /usr/local/google/home/pszczesniak/go/src/k8s.io/heapster/Godeps/_workspace/src/github.com/golang/glog/glog.go:410 +0x297

goroutine 19 [syscall]:
os/signal.loop()
        /usr/lib/google-golang/src/os/signal/signal_unix.go:22 +0x18
created by os/signal.init.1
        /usr/lib/google-golang/src/os/signal/signal_unix.go:28 +0x37
goroutine 21 [select]:
k8s.io/heapster/events/sources/kubernetes.(*KubernetesEventSource).watch(0xc820244990)
        /usr/local/google/home/pszczesniak/go/src/k8s.io/heapster/events/sources/kubernetes/kubernetes_source.go:101 +0x116e
created by k8s.io/heapster/events/sources/kubernetes.NewKubernetesSource
        /usr/local/google/home/pszczesniak/go/src/k8s.io/heapster/events/sources/kubernetes/kubernetes_source.go:160 +0x2cf
goroutine 33 [select]:
net/http.(*persistConn).writeLoop(0xc8200e3a20)
        /usr/lib/google-golang/src/net/http/transport.go:1009 +0x40c
created by net/http.(*Transport).dialConn
        /usr/lib/google-golang/src/net/http/transport.go:686 +0xc9d
goroutine 34 [IO wait]:
net.runtime_pollWait(0x7fd28259a440, 0x72, 0xc820014120)
        /usr/lib/google-golang/src/runtime/netpoll.go:157 +0x60
net.(*pollDesc).Wait(0xc820148ca0, 0x72, 0x0, 0x0)
        /usr/lib/google-golang/src/net/fd_poll_runtime.go:73 +0x3a
net.(*pollDesc).WaitRead(0xc820148ca0, 0x0, 0x0)
        /usr/lib/google-golang/src/net/fd_poll_runtime.go:78 +0x36
net.(*netFD).Read(0xc820148c40, 0xc820358000, 0x8000, 0x8000, 0x0, 0x7fd282594050, 0xc820014120)
        /usr/lib/google-golang/src/net/fd_unix.go:232 +0x23a
net.(*conn).Read(0xc8200261b8, 0xc820358000, 0x8000, 0x8000, 0x0, 0x0, 0x0)
        /usr/lib/google-golang/src/net/net.go:172 +0xe4
crypto/tls.(*block).readFromUntil(0xc820164840, 0x7fd28259a500, 0xc8200261b8, 0x5, 0x0, 0x0)
        /usr/lib/google-golang/src/crypto/tls/conn.go:455 +0xcc
crypto/tls.(*Conn).readRecord(0xc8200a6dc0, 0x1446317, 0x0, 0x0)
        /usr/lib/google-golang/src/crypto/tls/conn.go:540 +0x2d1
crypto/tls.(*Conn).Read(0xc8200a6dc0, 0xc82034e000, 0x1000, 0x1000, 0x0, 0x0, 0x0)
        /usr/lib/google-golang/src/crypto/tls/conn.go:901 +0x167
net/http.noteEOFReader.Read(0x7fd28259ab40, 0xc8200a6dc0, 0xc8200e3a78, 0xc82034e000, 0x1000, 0x1000, 0x0, 0x0, 0x0)
        /usr/lib/google-golang/src/net/http/transport.go:1370 +0x67
net/http.(*noteEOFReader).Read(0xc8201678c0, 0xc82034e000, 0x1000, 0x1000, 0xc82013d1c0, 0x0, 0x0)
        <autogenerated>:126 +0xd0
bufio.(*Reader).fill(0xc820019f20)
        /usr/lib/google-golang/src/bufio/bufio.go:97 +0x1e9
bufio.(*Reader).ReadSlice(0xc820019f20, 0xc82016910a, 0x0, 0x0, 0x0, 0x0, 0x0)
        /usr/lib/google-golang/src/bufio/bufio.go:328 +0x21a
net/http/internal.readLine(0xc820019f20, 0x0, 0x0, 0x0, 0x0, 0x0)
        /usr/lib/google-golang/src/net/http/internal/chunked.go:110 +0x4b
net/http/internal.(*chunkedReader).beginChunk(0xc820165740)
        /usr/lib/google-golang/src/net/http/internal/chunked.go:47 +0x39
net/http/internal.(*chunkedReader).Read(0xc820165740, 0xc820354601, 0x5ff, 0x5ff, 0x0, 0x0, 0x0)
        /usr/lib/google-golang/src/net/http/internal/chunked.go:77 +0xb7
net/http.(*body).readLocked(0xc82013d040, 0xc820354601, 0x5ff, 0x5ff, 0xc800000000, 0x0, 0x0)
        /usr/lib/google-golang/src/net/http/transfer.go:632 +0x98
net/http.(*body).Read(0xc82013d040, 0xc820354601, 0x5ff, 0x5ff, 0x0, 0x0, 0x0)
        /usr/lib/google-golang/src/net/http/transfer.go:624 +0x110
net/http.(*bodyEOFSignal).Read(0xc82013d080, 0xc820354601, 0x5ff, 0x5ff, 0x0, 0x0, 0x0)
        /usr/lib/google-golang/src/net/http/transport.go:1296 +0x26a
encoding/json.(*Decoder).refill(0xc820160340, 0x0, 0x0)
        /usr/lib/google-golang/src/encoding/json/stream.go:152 +0x287
encoding/json.(*Decoder).readValue(0xc820160340, 0x1, 0x0, 0x0)
        /usr/lib/google-golang/src/encoding/json/stream.go:128 +0x41b
encoding/json.(*Decoder).Decode(0xc820160340, 0xe0ece0, 0xc820165b30, 0x0, 0x0)
        /usr/lib/google-golang/src/encoding/json/stream.go:57 +0x159
k8s.io/kubernetes/pkg/watch/json.(*Decoder).Decode(0xc820165770, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0)
        /usr/local/google/home/pszczesniak/go/src/k8s.io/heapster/Godeps/_workspace/src/k8s.io/kubernetes/pkg/watch/json/decoder.go:50 +0xa5
k8s.io/kubernetes/pkg/watch.(*StreamWatcher).receive(0xc8201657a0)
        /usr/local/google/home/pszczesniak/go/src/k8s.io/heapster/Godeps/_workspace/src/k8s.io/kubernetes/pkg/watch/iowatcher.go:92 +0xf0
created by k8s.io/kubernetes/pkg/watch.NewStreamWatcher
        /usr/local/google/home/pszczesniak/go/src/k8s.io/heapster/Godeps/_workspace/src/k8s.io/kubernetes/pkg/watch/iowatcher.go:59 +0xbf
goroutine 16 [select]:
net/http.(*persistConn).readLoop(0xc8200e3a20)
        /usr/lib/google-golang/src/net/http/transport.go:976 +0xac7
created by net/http.(*Transport).dialConn
        /usr/lib/google-golang/src/net/http/transport.go:685 +0xc78
goroutine 27 [select]:
k8s.io/heapster/events/manager.(*realManager).Housekeep(0xc820245020)
        /usr/local/google/home/pszczesniak/go/src/k8s.io/heapster/events/manager/manager.go:64 +0x29f
created by k8s.io/heapster/events/manager.(*realManager).Start
        /usr/local/google/home/pszczesniak/go/src/k8s.io/heapster/events/manager/manager.go:49 +0x35
```
It's currently failing in gce-soak tests.

cc @fgrzadkowski @piosz

--- a/events/sinks/influxdb/influxdb.go
+++ b/events/sinks/influxdb/influxdb.go
@@ -115,6 +115,7 @@ func (sink *influxdbSink) ExportEvents(eventBatch *core.EventBatch) {
 func (sink *influxdbSink) sendData(dataPoints []influxdb.Point) {
 	if err := sink.createDatabase(); err != nil {
 		glog.Errorf("Failed to create infuxdb: %v", err)
+		return
 	}
 	bp := influxdb.BatchPoints{
 		Points:          dataPoints,

--- a/metrics/sinks/influxdb/influxdb.go
+++ b/metrics/sinks/influxdb/influxdb.go
@@ -124,6 +124,7 @@ func (sink *influxdbSink) ExportData(dataBatch *core.DataBatch) {
 func (sink *influxdbSink) sendData(dataPoints []influxdb.Point) {
 	if err := sink.createDatabase(); err != nil {
 		glog.Errorf("Failed to create infuxdb: %v", err)
+		return
 	}
 	bp := influxdb.BatchPoints{
 		Points:          dataPoints,

